vital/types/metadata.h

```diff
@@ -334,31 +347,46 @@ class typed_metadata
 class VITAL_EXPORT metadata
 {
 public:
+// The design for this collection requires that the elements in the
+// collection are owned by the collection and are only returned by
+// value. This is why the std::unique_ptr is used. Unfortunately, not
+// all stdc implementations support maps with unique_ptrs. The
+// following is done to work around this limitation.
 #ifdef VITAL_STD_MAP_UNIQUE_PTR_ALLOWED
-  typedef std::unique_ptr< metadata_item > item_ptr;
+  using item_ptr = std::unique_ptr< metadata_item >;
 #else
-  typedef std::shared_ptr< metadata_item > item_ptr;
+  using item_ptr = std::shared_ptr< metadata_item >;
 #endif
-  typedef std::map< vital_metadata_tag, item_ptr > metadata_map_t;
-  typedef metadata_map_t::const_iterator const_iterator_t;
+  using metadata_map_t = std::map< vital_metadata_tag, item_ptr >;
+  using const_iterator_t = metadata_map_t::const_iterator;
 
   metadata();
   ~metadata();
 
-
   /// Add metadata item to collection.
   /**
-   * This method adds a metadata item to the collection. The collection
-   * takes ownership of the item and managed the memory.
+   * This method adds a metadata item to the collection. The
+   * collection takes ownership of the item and managed the memory.
    *
-   * @param item New metadata item to be copied into collection.
+   * @param item New metadata item to be added to collection.
    */
   void add( metadata_item* item );
 
 
+  /// Add metadata item to collection.
+  /**
+   * This method adds a metadata item to the collection. The
+   * collection takes ownership of a copy of the item and managed the
+   * memory.
+   *
+   * @param item New metadata item to be added to collection.
+   */
+  void add( std::shared_ptr<metadata_item const> const& item );
+
+
   /// Remove metadata item.
   /**
-   * The metadata item that corresponds with the tag is deleted it it
+   * The metadata item that corresponds with the tag is deleted it
```

Comment:
+1 getting rid of duplicate word. -1 not noticing that "it" should be "if" :wink:.